### PR TITLE
Document Top-Level Subdirectories

### LINKED ISSUE #83: Add short directory descriptions to the readme
## ⚠️ Is your feature request related to a problem? Please describe
No clue why there are so many directories and what they do.

## 💡 Describe the solution you'd like
Add directory descriptions to readme or docs.

## 🤚 Do you want to develop this feature yourself?

- [X] Yes
- [X] No

### PR DESCRIPTION
**Pull request by adamaveray**
## 🚀 Description

Adds a brief description of each top-level subdirectory in the project (Closes #83)

## 📄 Motivation and Context

As requested in #83, explains all the root directories in the project since the two libraries and config for the linting tools makes the project structure much larger than a default new Android app.

## 🧪 How Has This Been Tested?

(Documentation-only change – no code changes)

## 📷 Screenshots (if appropriate)

(N/A)

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,25 @@ If you're using [JitPack](https://jitpack.io/), you don't need any further confi
...
+
+The project includes three sub-projects, each in their own subdirectories:
+
+- **`app`:** The source for the final Android application.
+- **`library-android`:** The source for an Android library including UI.
+- **`library-kotlin`:** The source for a UI-less Kotlin library.
+
+The following additional top-level directories configure & support building the app & projects:
+
+- **`buildSrc`:** Builds any shared plugins, tasks, etc used when building the primary sub-project and loads the Gradle Version Catalog [(see more)](https://stackoverflow.com/a/13875350).
```
Comment (cortinico) [inline README.md:98]:
```suggestion
- **`buildSrc`:** Contains shared Gradle logic as [precompiled script plugins](https://docs.gradle.org/current/userguide/custom_plugins.html#sec:precompiled_plugins)```

File: README.md [hunk 2]
```diff
@@ -85,6 +85,25 @@ If you're using [JitPack](https://jitpack.io/), you don't need any further confi
...
+- **`config`:** Contains the [Detekt configuration file](https://detekt.dev/docs/introduction/configurations/).
+- **`gradle`:** Contains the Gradle Version Catalog and the [Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html).
```
Comment (cortinico) [inline README.md:100]:
```suggestion
- **`gradle`:** Contains Gradle Configuration files such as the Gradle Version Catalog and the [Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html).
```

**Review by cortinico — APPROVED** [review]
Great stuff, left a couple of corrections

**Comment by adamaveray** [comment]
@cortinico Thank you for your suggestions! Makes it much clearer. Those updates are now in.

**Review by cortinico — APPROVED** [review]

**Comment by adamaveray** [comment]
Not sure why the Windows check failed...

**Comment by cortinico** [comment]
I've retriggered

**Pull request merged**